meteor-slingshot/lib/upload.js

```diff
@@ -202,13 +202,22 @@ Slingshot.Upload = function (directive, metaData) {
         );
       });
 
-      xhr.open('POST', self.instructions.upload, true);
+      let isR2 = self.instructions.upload.includes('r2.cloudflarestorage.com');
+      let method = 'POST';
```

Comment:
`const method = isR2 ? 'PUT' : 'POST'`

Reply:
fixed

Reply:
Added a "service" key as optional to pass this parameter to upload function.